app/controllers/order_items_controller.rb

```diff
@@ -0,0 +1,70 @@
+class OrderItemsController < ApplicationController
+  # before_action :set_order_item, only: [:show, :edit, :update, :destroy]
+  #
+  def index
+    @user = User.find(params[:user_id])
+    @orders = @user.orders
+  end
+  #
+  # def show
+  #   @user = User.find(params[:user_id])
+  # end
+  #
+  # def new
+  #   @order_item = OrderItem.new
+  #   @order = Order.find(params[:order_id])
+  #   @user = User.find(params[:user_id])
+  # end
+  #
+  # def edit
+  #   @user = User.find(params[:user_id])
+  #   @order = Order.find(params[:order_id])
+  #   @user = User.find(params[:user_id])
+  # end
+  #
+  # def create
+  #   @user = User.find(params[:user_id])
+  #   @order = Order.find(params[:order_id])
+  #   @order_item = OrderItem.new(order_item_params)
+  #     if @order_item.save
+  #       redirect_to user_order_order_items_path(@user,@order.id)
+  #     else
+  #       render "new"
+  #     end
+  # end
+
+  def shipped
+    # user = User.find(params[:user_id])
+    order_item = OrderItem.find(params[:id])
+    if !order_item.shipped
+      order_item.update(:shipped => true)
+      Order.check_order_shipped(order_item)
+    else
+      order_item.update(:shipped => false)
+      Order.check_order_shipped(order_item)
```

Comment:
Since we're calling the `check_order_shipped` method regardless of whether we marked/unmarked the present order item as shipped, we can move this outside of the `if / else` statement entirely and only need to write it once.